### DISCUSSION TITLE
Multi-variate multi-step forecasting

### DISCUSSION THREAD

**Discussion by jSwords91** [ourownstory/neural_prophet#592] [Q&A - forecasting best practices] [answered]
Hi. 
I have a dataset with columns: `"ds", "y", "temperature", "rainfall"`. 
Temperature and rainfall are included in the dataset history, and they also have forecasted values from t through to t+7 days. `y` is of course `NaN` for these forecasted dates.

I am having trouble training a multi-variate model and predicting in to the future up to t+7 days, using the forecasted weather values. I can predict the historic data fine (using this notebook as reference: 
[https://github.com/ourownstory/neural_prophet/blob/main/tutorials/feature-use/lagged_covariates_energy_ercot.ipynb](url)

Do you have any advice how we can do this?

**Comment by jSwords91** [comment, marked as answer]
Self-solved...

This notebook is helpful:
[https://github.com/ourownstory/neural_prophet/blob/main/docs/source/future-regressors.md](url) 

However the crucial element missing is that in the `make_future_dataframe `call there is a `regressors_df `option which is by default `None`. If you change this to be your dataframe of future regressors (ie. `regressors_df=future_regressors_df`), it will work. 

In addition, I would advise setting `periods=len(future_regressors_df)` that way you'll be predicting only as far as your future regressors allow.

Great library  - loving it so far.

**Comment by neomatrix369** [comment]
The above url does not work when we click on it, the URL to click on is [https://github.com/ourownstory/neural_prophet/blob/main/docs/source/future-regressors.md](https://github.com/ourownstory/neural_prophet/blob/main/docs/source/future-regressors.md)

**Reply by jSwords91** [reply]
It must have changed. Thanks for finding the new location!
